### DISCUSSION TITLE
Local execution monitoring

### DISCUSSION THREAD

**Discussion by tiborsekera** [elyra-ai/elyra#1478] [Q&A]
Are there any plans to support:
- monitoring of the elyra pipeline, when running locally?
- command line interface, so that one can run pipeline via bash script or so? 

 Or, is any of that already possible?

**Comment by kevin-bates** [comment]
Hi @tiborsekera - thanks for your questions.

Pull request #1246 will address the second item and is nearly ready.  If you'd like to take that branch for a spin, that would be helpful.  

We might need further information as to what kind of _monitoring_ you'd like relative to your first item.

**Reply by tiborsekera** [reply]
Hi @kevin-bates !

Thank you for the reply!

Reg. monitoring:
Currently, to now that the locally executed pipeline is running, which step is running, etc. I have to use some custom logging and other libraries. It would be great to be able to see, at least via command line, that the pipeline is running. Even better, to have some elyra_pipeline.log file, where one can see whether the pipeline is running, crashed, which step is running, etc.

**Reply by kevin-bates** [reply]
Thanks.  I think it would be good to define what _local execution_ really means.  I believe this should be viewed as a means of prototyping the pipeline and primarily a development aid.  However, it does seem like an _execution ledger_ could be something useful that would contain node execution timings and failure information pertaining to the executed pipeline.

Definitely feels like something we should discuss further.

Does anyone else have ideas of where we should head with this?
